portfolio/src/main/webapp/gallery-script.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Load project details from local registry of projects (projects.json)
+ */
+const loadProjects = async() => {
+  const request = new XMLHttpRequest();
+  request.open("GET", "projects.json", true);
+  request.onload = () => addProjectsToDOM(request.responseText);
+  request.onerror = () => console.error(request.statusText);
+  request.send(null);
+}
+
+/**
+ * Handler for the successful retrieval of project details
+ * from the local project registry
+ */
+const addProjectsToDOM = async (json) => {
+  /*
+    json will always contain a top-level object which has a single member.
+    this member will be an array of objects. these objects will have three properties:
+    - title : a string containing the title of the project
+    - URL : a string which contains a URL linking to the blog post about
+        the project
+    - preview : a string which contains a URL linking to the preview image
+        for this project
+  */
+  const data = JSON.parse(json);
+  data.projects.map(async (project) => {
```

Comment:
I am not clear on what are you trying to achieve here. You are trying to Map the json data to what? It's not easily readable.

Reply:
I've updated the docs now 👍 https://github.com/thomasogara/step-repository/pull/2/commits/5a27a88da131033ecf7d5279c4715f4b574eebdd